Training/Training/main.c

```diff
@@ -0,0 +1,176 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// main.c
+// Program to convert and display decimal value into binary and hexadecimal value.
+// ------------------------------------------------------------------------------------------------
+
+#include<stdio.h>
+#include<malloc.h>
+#include<math.h>
+
+int mCount;
+
+//Function for counting the number of digits
+int Count (int dec, int div) {
+   int c = 0;
+   while (dec > 0) {
+      dec /= div;
+      c++;
+   }
+   return c;
+}
+
+int* DecToBin (int dec) {
+   mCount = 0;
+   int decReal = dec;
+   if (dec < 0) dec *= -1;
+   int decTemp = dec;
+   //Counting the number of digits in binary representation
+   mCount = Count (dec, 2);
+   // Representing binary in multiples of 8-bits accordingly
+   if (mCount % 8 != 0) mCount += (8 - (mCount % 8));
+   int* binRev = NULL;
+   do {
+      binRev = (int*)malloc (mCount * sizeof (int));
+   } while (binRev == NULL);
+   //Storing the remainders into binRev
+   for (int i = 0; i < mCount; i++) {
+      binRev[i] = decTemp % 2;
+      decTemp /= 2;
+   }
+   if (decReal > 0) {
+      int* bin;
+      do {
+         bin = (int*)malloc (mCount * sizeof (int));
+      } while (bin == NULL);
+      // Reverse storing the stored remainders
+      for (int j = 0; j < mCount; j++)
+         bin[j] = binRev[mCount - j - 1];
+      return bin;
+   } else { //Performing two's compliment

```

Comment:
Also, it's nice that you've tried to implement 2's complement logic manually, but there is an easier way to do this. Explore bit shift operators as an alternative.